### PR TITLE
Log instead of error out when process-agent is not enabled in config in MacOS

### PR DESCRIPTION
**Pull request by shang-wang**
### What does this PR do?

Currently the process corecheck returns an error when process-agent is disabled in the config. This results in process-agent showing up in error status in datadog-agent UI. The PR logs the message instead and return no error, so it shouldn't confuse customers anymore.

cc: @DataDog/burrito 

### Motivation

Confusing status page.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/collector/corechecks/embed/process_agent.go [hunk 1]
```diff
@@ -129,7 +129,8 @@ func (c *ProcessAgentCheck) run() error {
 func (c *ProcessAgentCheck) Configure(data integration.Data, initConfig integration.Data, source string) error {
 	// handle the case when the agent is disabled via the old `datadog.conf` file
 	if enabled := config.Datadog.GetBool("process_config.enabled"); !enabled {
```
Comment (leeavital) [inline pkg/collector/corechecks/embed/process_agent.go]: Will it be a problem that things like `source/binPath` won't be set for calls to `run()` later?

Comment (shang-wang) [reply pkg/collector/corechecks/embed/process_agent.go]: You are right, I removed the early return. Thanks.

**Review by leeavital — APPROVED** [review]

File: pkg/collector/corechecks/embed/process_agent.go [hunk 2]
```diff
@@ -129,7 +129,7 @@ func (c *ProcessAgentCheck) run() error {
...
-		return fmt.Errorf("Process Agent disabled through main configuration file")
+		log.Info("Process Agent disabled through main configuration file")
```
Comment (hush-hush) [inline pkg/collector/corechecks/embed/process_agent.go]: Shouldn't we return nil here ?

Comment (truthbk) [reply pkg/collector/corechecks/embed/process_agent.go]: +1

Comment (shang-wang) [reply pkg/collector/corechecks/embed/process_agent.go]: I think this function is only meant to parse all the configs. The idea is to always let process-agent run and use the config to stop process-agent once it's started. If we break early then there are several things that didn't get initialized(like binPath) so when process-agent gets running there'll be one or two unnecessary(non critical) errors(like fork error).

Comment (shang-wang) [reply pkg/collector/corechecks/embed/process_agent.go]: If `binPath` is not initialized you'll see error:
```
2019-09-03 11:35:50 EDT | CORE | ERROR | (pkg/collector/runner/runner.go:292 in work) | Error running check Process Agent: fork/exec : no such file or directory
```

**Review by truthbk — APPROVED** [review]
You might want to add a comment to explain why we still initialize everything even when `process_config.enabled` is false. But otherwise, this looks good to me.

File: pkg/collector/corechecks/embed/process_agent.go (hunk 2 above)
Comment (albertvaka) [reply pkg/collector/corechecks/embed/process_agent.go]: Merging to build new RC, but would be nice to add this explanation as a comment in the code, as jaime suggested (on a new PR).

**Pull request merged**